avocado/core/test.py

```diff
@@ -239,6 +242,10 @@ def record_and_warn(*args, **kwargs):
         self.time_elapsed = -1
         unittest.TestCase.__init__(self, methodName=methodName)
 
+        self.teststmpdir = None
+        if tests_tmpdir is not None:
+            self.teststmpdir = tests_tmpdir
```

Comment:
One question here. Do we expect people to check, whether `teststmpdir` is available or not? From my understanding of the usage I'd expect the tests which need it to depend on it blindly. If that's true it'd be better to raise exception when teststmpdir is not available (which could be implemented in the suggested property).

If we think many users wanted to write tests which treats non-existing `teststmpdir`, then we should keep returning None so they can check `self.workdir if self.teststmpdir is None else self.teststmpdir`. But I doubt many users would want to write such tests.

There could be also another "optimal" variant, which would assume that if `teststmpdir` is not provided by runner, `teststmpdir` is actually `self.workdir`. That would allow tests to always work, only sometimes without sharing the data. For some tests this would be the simplest solution for test writers, but I consider it magic and ugly.